help/learn/Custom-Machine-Learning.md

```diff
@@ -0,0 +1,132 @@
+# Custom ML models
+ 
+ Datagrok supports three ML engines out of the box: H2O, Caret, Chemprop. In addition, the platform allows users to build their own customizable ML models. This capability provides full access and control to algorithms available in ML libraries in any of the supported languages (such as Python, R, and Julia). The users may construct and configure any chosen model in their custom data pipeline.   
+ 
+ Custom model functionality is implemented as a two-step process using train and apply functions. Train function is used to build/train a model based on provided features and a target variable. The trained model is then stored as an object inside a given directory. Apply function receives previously saved model and applies it to provided feature columns. The resulting predictions are returned in a form of a single column in a dataframe.  
+ 
+Custom models utilize Predictive Model interface alongside other out-of-the-box solutions. Once implemented, custom models can be chosen from the list of model engines.
+
+
+## Train
+ 
+Header Parameters
+
+- #meta.mlname: CustomName – Name of the custom train function.
```

Comment:
Explanations for parameters will be a bit easier to read if we mark them as code:
\`#meta.mlname: CustomName\` – Name of the custom train function
It will give us a similar look:
`#meta.mlname: CustomName` – Name of the custom train function
Periods at the end of list items can be omitted.